aoc2022/src/main/scala/day15.scala

```diff
@@ -0,0 +1,198 @@
+ * 
+ * 
+ *   ┌──────────────────────────────────────────┐
+ *   │                 . .                      │
+ *   │                  X                       │
+ *   │                 . .                      │
+ *   │            .   .   .                     │
+ *   │             . .     .                    │
+ *   │              X       .                   │
+ *   │             . .       .   .              │
+ *   │            A   .       . .               │
+ *   │           .#.   .       X                │
+ *   │          .###.   .     . .               │
+ *   │         .#####.   .   .   .              │
+ *   │  .     .#######.   . .     .             │
+ *   │   .   .#########.   X       .            │
+ *   │    . .###########. . .       .           │
+ *   │     D######S######B   .       .          │
+ *   │    . .###########.     .     .#.         │
+ *   │   .   .#########.       .   .###.   .    │
+ *   │  .     .#######.         . .#####. .     │
+ *   │         .#####.           .#######.      │
+ *   │          .###.             .#####. .     │
+ *   │           .#.               .###.   .    │
+ *   │            C                 .#.         │
+ *   │           . .                 .          │
+ *   │          .   .               . .         │
+ *   │         .     .             .   .        │
+ *   │                                          │
+ *   └──────────────────────────────────────────┘
+ * 
+ *            1         2
+ *   0        0         0   x
+ *  0┌────────────────────────┐
+ *   │S########S#############S│
+ *   │#############R####B#####│
+ *   │########################│
+ *   │########################│
+ *   │#####B##################│
+ *   │##################F#####│
+ *   │##############.#########│
+ *   │########################│
+ *   │########################│
+ * 10│########W######Q########│
+ *   │#################B######│
+ *   │##B#####################│
+ *   │########################│
+ *   │########################│
+ *  y│S######################S│
+ *   └────────────────────────┘
+ * 
+ * The solution involves checking for the computed line intersections XY coordinates if it's within the range of a 
+ * sensor. If not, then that must be the distress beacon. 
+ *
+ */
+
+
+object day15 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  private val input: Vector[Sensor] =
+    
+    def parseSensor(s: String): Sensor = s match
+      case s"Sensor at x=$x, y=$y: closest beacon is at x=$bx, y=$by" => 
+        val sensor: Point = Point(x.toInt, y.toInt)
+        val beacon: Point = Point(bx.toInt, by.toInt)
+        Sensor(sensor, beacon, sensor.manhattan(beacon))
+      case _ => sys.error(s"Couldn't parse sensor: $s")
+    
+    Source
+      .fromResource(s"day$day.txt")
+      .getLines
+      .map(parseSensor)
+      .toVector
+    
+  case class Sensor(loc: Point, beacon: Point, innerDist: Int)
```

Comment:
Instead of `innerDist` being a parameter, I think it is better to declare a `val innerDist` within the case class; that way there can be no "illegal" combination of values.

Reply:
Nice one